Fix/jeongsoo title&badge

Delcastin opened the pull request:
칭호 & 뱃지 기능 분기 로직 및 추가로 적용된 칭호들을 추가드립니다.

- NONE("칭호 없음", "기본값", TitleCategory.SYSTEM)
- FIRST_ROOM_ENTER("첫 입장", "처음 방에 입장했을 때 취득", TitleCategory.SYSTEM), 
- CLEAN_HUNTER("청결 헌터", "신고5회 이상 시 취득", TitleCategory.SYSTEM)
- FROM_9_START("From 9 Start", "매일 오전9시 정각에 접속, 7일 이상 진행 시 취득", TitleCategory.ATTENDANCE),
 
// ⏱ 집중 관련
- FOCUS_BEGINNER("Focus Beginner", "하루30분 이상 집중1회", TitleCategory.FOCUS), 
- FOCUS_RUNNER("Focus Runner", "하루1시간 이상 집중, 3일 연속", TitleCategory.FOCUS), 
- FIRST_ONE_HOUR("첫 걸음도 한 시간부터!", "누적 집중1시간 이상", TitleCategory.FOCUS), 
- HUNDRED_FOCUS("Hundred Focus", "누적 집중100시간 달성", TitleCategory.FOCUS), 
- TEN_HOURS_ONE_DAY("Only Running for day", "하루에10시간 이상 집중했을 시 취득", TitleCategory.FOCUS), 

// 💬 참여/채팅
- CHATTER("Wide Foot", "채팅100회 이상", TitleCategory.CHAT),

// 📖 수집
- TITLE_COLLECTOR("칭호 수첩", "칭호10개 이상 획득", TitleCategory.SPECIAL);earnedTitleCount - UserActivity

추가된 칭호들은 상기와 같습니다! 또한, 추가로 변경사항이 있는데,
최근에 develop branch를 보니 PATCHMAPPING을 사용한 것을 확인했습니다. 그걸 확인하고, CORS 설정 중에 PATCH 설정도 같이 추가 드렸으니, 참고해주시길 부탁 드립니다!

Delcastin committed:
fix : jeongsoo-title & badge

변경사항
-. 칭호의 구성에 따라 LoginHistory, RoomHistory Domain 추가
-. Repository들 추가
-. 로직도 분리 로직에 따라 변경할 예정.

Delcastin committed:
fix : jeongsoo-title & badge

변경사항
-. 칭호 구성 완료, 기능들 테스트까지 완료하였습니다.
-. Test는 아직 진행이 안 되었으니 이 점까지만 진행하고 Commit 진행합니다.

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

추가 변경사항
-. 9시에 로그인했는지 여부를 파악해서 칭호를 부여하는 로직 추가
-. Controller Test, From9Start 칭호 부여 로직 Service 단위 Test 추가

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

추가 변경사항
-. 9시에 로그인했는지 여부를 파악해서 칭호를 부여하는 로직 추가
-. Controller Test, From9Start 칭호 부여 로직 Service 단위 Test 추가

Delcastin committed:
Merge branch 'develop' into fix/jeongsoo-title&badge

sksqlrlek commented on livestudy/src/main/java/org/livestudy/controller/TitleController.java:
```diff
@@ -30,9 +33,28 @@ public class TitleController {
+    public ResponseEntity<UserTitleResponse> equipTitle(
+            @Parameter(description = "사용자 ID", example = "1")@RequestParam Long titleId,
+            @Parameter(description = "장착할 칭호 ID", example = "5")@PathVariable Long userId
+    ) {
```
@Parameter(description = "사용자 ID", example = "1")@RequestParam Long titleId

이렇게 되어 있는데, 아래 장착할 칭호 id 설명과 바뀐 거 같습니다

Delcastin replied:
아, 이 부분은 바꾸겠습니다!

sksqlrlek reviewed (commented):
확인했습니다.

sksqlrlek commented on livestudy/src/main/java/org/livestudy/repository/factory/UserChatRepository.java:
이거 livestudy/src/main/java/org/livestudy/repository에 있는 ChatRepository 있는 것과 겹치는 내용이라 하나로 합치는 게 좋아보입니다.

Delcastin replied:
알겠습니다, 이 부분은 합치도록 할께요!

sksqlrlek commented on livestudy/src/main/java/org/livestudy/repository/factory/UserReportRepository.java:
이거 livestudy/src/main/java/org/livestudy/repository에 있는 ReportRepository 있는 것과 겹치는 내용이라 하나로 합치는 게 좋아보입니다.

Delcastin replied:
네, 이 부분은 합치도록 하겠습니다!

sksqlrlek reviewed (commented):
추가 리뷰입니다.

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

추가 변경사항
-. 9시에 로그인했는지 여부를 파악해서 칭호를 부여하는 로직 추가
-. Controller Test, From9Start 칭호 부여 로직 Service 단위 Test 추가

Delcastin committed:
Merge remote-tracking branch 'origin/fix/jeongsoo-title&badge' into fix/jeongsoo-title&badge

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

추가 변경사항
-. 9시에 로그인했는지 여부를 파악해서 칭호를 부여하는 로직 추가
-. Controller Test, From9Start 칭호 부여 로직 Service 단위 Test 추가

- 현 커밋에서는 Github Feedback 내용, UserReportRepository.java & ReportRepository.java 병합,
UserChatRepository.java & ChatRepository.java 병합

- TitleController에 있는 titleId에 대한 parameter 설명 수정

Delcastin committed:
Fix : jeongsoo-title&badge 기능 수정

-. 하기의 칭호 및 뱃지 부여 기능에 대해서 분기 로직 및 조건 충족 로직을 리팩토링하여 변경 드립니다.

- 현재 맞춰진 조건들에 대해 parameter당 한 개씩 단위 테스트를 작성하였습니다.

-. 조건의 종류는 하기와 같습니다.
1. 칭호의 갯수
2. 하루에 몇 분이상 집중했는지
3. 며칠동안 하루에 몇 분이상 집중했는지
4. 누적 집중 시간
5. 누적 채팅 횟수
6. 처음 방에 입장했을 때
7. 특정 기간동안 빠지지 않고 접속했을 때
등등의 조건을 기준으로 칭호 & 뱃지 기능을 부여하는 로직을 만들었습니다.

Teams 변경사항 고지
- 최근 develop Branch 관련하여 소통하였을 때 PatchMapping을 쓰셨다는 말씀을 들었습니다.
- 확인하여, CORS 설정에 "PATCH" 설정을 추가드리오니, 이 점 확인해주시길 부탁 드리겠습니다.

추가 변경사항
-. 9시에 로그인했는지 여부를 파악해서 칭호를 부여하는 로직 추가
-. Controller Test, From9Start 칭호 부여 로직 Service 단위 Test 추가

- 현 커밋에서는 Github Feedback 내용, UserReportRepository.java & ReportRepository.java 병합,
UserChatRepository.java & ChatRepository.java 병합

- TitleController에 있는 titleId에 대한 parameter 설명 수정

- RoomHistory & LoginHistory Entity의 연관관계 형성 및 이를 토대로 한 Test 변경 및 진행 완료하였습니다.
